x/ccv/provider/keeper/keeper.go

```diff
@@ -306,8 +318,38 @@ func (k Keeper) SetUnbondingOpIndex(ctx sdk.Context, chainID string, valsetUpdat
 	store.Set(types.UnbondingOpIndexKey(chainID, valsetUpdateID), bz)
 }
 
-// This index allows retreiving UnbondingDelegationEntries by chainID and valsetUpdateID
-func (k Keeper) GetUnbodingOpIndex(ctx sdk.Context, chainID string, valsetUpdateID uint64) ([]uint64, bool) {
+// IterateOverUnbondingOpIndex iterates over the unbonding indexes for a given chain id.
+func (k Keeper) IterateOverUnbondingOpIndex(ctx sdk.Context, chainID string, cb func(vscID uint64, ubdIndex []uint64) bool) {
+	store := ctx.KVStore(k.storeKey)
+	prefix := append(types.HashString(types.UnbondingOpIndexPrefix), types.HashString(chainID)...)
+	iterator := sdk.KVStorePrefixIterator(store, prefix)
+
+	defer iterator.Close()
+	for ; iterator.Valid(); iterator.Next() {
+		// parse key to get the current VSC ID
+		var vscID uint64
+		vscBytes := iterator.Key()[64:]
```

Comment:
Why `[64:]`? The key's format should be `UnbondingOpIndexPrefix | chainID | vscID`. Shouldn't this be `iterator.Key()[len(iterator.Key())-64:]` instead. Is this because `HashString()` returns a 32 byte hash? We should make sure that https://github.com/cosmos/interchain-security/issues/18 is not going to break this logic.

Reply:
The three key's parts in `UnbondingOpIndexPrefix | chainID | vscID` are hashed into 32 bytes fixed output.
https://github.com/cosmos/interchain-security/blob/579db08d6bb34ea2f5ad793841f66f464935f6bb/x/ccv/provider/types/keys.go#L117

So `iterator.Key()[64:]` returns the key's tail containing a `vscID` digest
 while `iterator.Key()[len(iterator.Key())-64:]` would return the aggregated `chainID | vscID` hashes.

I believe we're safe regarding #18 since we use static length keys, yet its worth having a second opinion on this.

Reply:
Let's not hardcode the `64` here. Use helper function and the length of the variable. See the standard way to do this in ibc-go:

https://github.com/cosmos/ibc-go/blob/main/modules/apps/29-fee/types/keys.go